src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -593,8 +594,8 @@ public function install_package( $args = array() ) {
 			}
 		}
 
-		// Copy new version of item into place.
-		$result = copy_dir( $source, $remote_destination );
+		// Move new version of directory into place.
+		$result = move_dir( $source, $remote_destination, true );
 
 		// Clear the working folder?
 		if ( $args['clear_working'] ) {
```

Comment:
```suggestion
		// Put the new version of directory into place.
		if ( $args['clear_working'] ) {
			$result = move_dir( $source, $remote_destination, true );
		} else {
			$result = copy_dir( $source, $remote_destination );
		}

		// Clear the working folder (accounts for `move_dir()` falling back to `copy_dir`)?
		if ( $args['clear_working'] ) {
```

While testing, I noticed that if the clear_working flag is false the files are retained in the upgrade directory so if that is the case then this will need to use `copy_dir()`.

For testing I forced it hacking it to false after this line
https://github.com/WordPress/wordpress-develop/blob/5836a438084356ebd232ffc152e18a6cf1c4f16f/src/wp-admin/includes/class-wp-upgrader.php#L466-L467

Reply:
Ugh, I forgot to say: I actually think the call to `::delete` can be within the same condition as `move_dir()` but GH won't let me make suggestions outside the diff.